### PR TITLE
feat!: Notify Operator via LSP1 in LSP7 and LSP8

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?
## 🚀 Feature

- Notify Operator via LSP1 in LSP7 and LSP8

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP7DigitalAsset/ILSP7DigitalAsset.sol [hunk 1]
```diff
@@ -41,15 +41,20 @@ interface ILSP7DigitalAsset is IERC165, IERC725Y {
...
+        bytes operatorNotificationData
     );
 
     /**
      * @dev Emitted when `tokenOwner` disables `operator` for `amount` tokens and set its {`authorizedAmountFor(...)`} to `0`.
      * @param operator The address revoked from operating
      * @param tokenOwner The token owner
      */
-    event RevokedOperator(address indexed operator, address indexed tokenOwner);
+    event RevokedOperator(
```
Comment (b00ste) [inline contracts/LSP7DigitalAsset/ILSP7DigitalAsset.sol:53]: Add `@param` description for `operatorNotificationData`

File: contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol [hunk 2]
```diff
@@ -97,16 +98,34 @@ abstract contract LSP7DigitalAssetCore is ILSP7DigitalAsset {
...
+        bytes memory operatorNotificationData
+    ) public virtual {
+        _updateOperator(msg.sender, operator, 0, operatorNotificationData);
+
+        bytes memory lsp1Data = abi.encode(
+            msg.sender,
+            0,
+            operatorNotificationData
+        );
+        _notifyTokenOperator(operator, lsp1Data);
```
Comment (b00ste) [inline contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol:128]: Shouldn't we use different type id for authorising operator and revoking operator?

**Review by b00ste — APPROVED** [review]

File: contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol (hunk 2 above)
Comment (b00ste) [reply contracts/LSP7DigitalAsset/LSP7DigitalAssetCore.sol:128]: To be discussed in the next SC meeting.

File: contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAssetCore.sol [hunk 3]
```diff
@@ -473,6 +509,34 @@ abstract contract LSP8IdentifiableDigitalAssetCore is
...
+        address operator,
+        bytes memory lsp1Data
+    ) internal virtual {
+        if (
+            ERC165Checker.supportsERC165InterfaceUnchecked(
+                operator,
+                _INTERFACEID_LSP1
+            )
+        ) {
+            (bool sucess, bytes memory returnData) = operator.call(
```
Comment (skimaharvey) [inline contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAssetCore.sol:530]: solhint

**Review by skimaharvey — APPROVED** [review]

**Pull request merged**